symplyphysics/laws/waves/wavelength_of_standing_wave_in_string_with_fixed_ends.py

```diff
@@ -19,13 +22,38 @@
 ## lambda - wavelength of standing wave
 ## L - length of the string
 
-integer_factor = Symbol("integer_factor", dimensionless, positive=True)
+integer_factor = Symbol("integer_factor", dimensionless, integer=True, positive=True)
 wavelength = Symbol("wavelength", units.length, positive=True)
 string_length = Symbol("string_length", units.length, positive=True)
 
 law = Eq(integer_factor * wavelength / 2, string_length)
 
-# TODO: derive from boundary condition `u(L, t) = 0`
+# Derive from boundary condition `u(L, t) = 0`
+
+_wavenumber = wavenumber_def.definition.rhs.subs(
+    wavenumber_def.wavelength, wavelength,
+)
+
+_standing_wave = standing_wave_law.law.rhs.subs(
+    standing_wave_law.angular_wavenumber, _wavenumber,
+)
+
+_standing_wave_at_string_end = _standing_wave.subs(
+    standing_wave_law.position, string_length,
+)
+
+_wavelength_solution = solve(_standing_wave_at_string_end, wavelength)[0]
+
+# We can show that any wavelength that is a whole number of times shorter than the
+# one we just got as the solution of the equation is also a resonant wavelength.
+
+_wavelength_solution_set = _wavelength_solution / integer_factor
```

Comment:
Would be great if we could somehow tweak trigonometric functions to return a general solution, say, $`x = \pi + 2 \pi n`$ or something

Reply:
I will look into it.